src/cpp/shared/combat/army.cpp

```diff
@@ -1343,83 +1345,100 @@ void SpecialAttackGraphics(army *attacker, army *target) {
   gpCombatManager->limitCreature[attacker->owningSide][attacker->stackIdx]++;
   gpCombatManager->DrawFrame(0, 1, 0, 1, 75, 1, 1);
 
-  int targMidX = target->MidX();
-  int targMidY = target->MidY();
-  if (attacker->creatureIdx == CREATURE_LICH || attacker->creatureIdx == CREATURE_POWER_LICH) {
-    targMidX = gpCombatManager->combatGrid[target->occupiedHex].centerX;
-    targMidY = gpCombatManager->combatGrid[target->occupiedHex].occupyingCreatureBottomY - 17;
-  }
-  int projStartX;
-  if (attacker->facingRight == 1)
-    projStartX = attacker->frameInfo.projectileStartOffset[1][0] + gpCombatManager->combatGrid[attacker->occupiedHex].centerX;
-  else
-    projStartX = gpCombatManager->combatGrid[attacker->occupiedHex].centerX - attacker->frameInfo.projectileStartOffset[1][0];
-  int projStartY = attacker->frameInfo.projectileStartOffset[1][1] + gpCombatManager->combatGrid[attacker->occupiedHex].occupyingCreatureBottomY;
-  int totXDiff = targMidX - projStartX;
-  char firingLeft = 0;
-  if (targMidX - projStartX < 0) {
-    firingLeft = 1;
-    totXDiff = -totXDiff;
+  int targMidX, targMidY;
+  int totXDiff, yDiff;
+  char firingLeft;
+  {
```

Comment:
Why do you use scope when there is no conditional statement?

Reply:
This is a technique I've become a fan of. It's a nice way to separate sections of the function without breaking things out into new functions. 

I don't remember where exactly I learned it, but http://number-none.com/blow/john_carmack_on_inlined_code.html played an influence.